packages/api/src/router/workspace.ts

```diff
@@ -86,6 +87,30 @@ const inviteRouter = createTRPCRouter({
       ),
 
     create: protectedProcedure
+      .meta({
+        authorizationCheck: ({ canUser, input }) =>
+          canUser
+            .perform(Permission.IamSetPolicy)
+            .on({ type: "workspace", id: input.workspaceId }),
+      })
+      .input(
+        z.object({
+          name: z.string(),
+          description: z.string().optional(),
+        }),
+      )
+      .mutation(async ({ ctx, input }) =>
+        ctx.db
+          .insert(workspace)
+          .values({
+            name: input.name,
+            slug: input.name,
+          })
+          .returning()
+          .then(takeFirst),
+      ),
```

Comment:
Don't we already have a function to creat workspaces?

Reply:
It was coupled to the invite flow in a really odd way.

Reply:
On a fresh instance, I couldn't get in.